### ISSUE TITLE
Syntax sugar for no-query-result condition

### ISSUE THREAD

**Issue by traut** [blackstork-io/blackstork-cli#28]
## Background

https://github.com/blackstork-io/fabric/issues/142 introduces `dynamic` blocks. One typical pattern is to select which content to render based on the query result.

## Design

Implement syntactic sugar t
```hcl

content text {
  query = ".data.inline.my_elements"
  value = "The elements: {{ .data.inline.my_elements }}"

  when_empty_result content text {
    ...
  }
}


```
is unpacked into
```hcl
dynamic content text {
  query = ".data.inline.my_elements"
  condition_query = "(.query_result == {} or .query_result == [] or .query_result == null) | not"

  value = "The elements: {{ .data.inline.my_elements }}"
}

dynamic content text {
  query = ".data.inline.my_elements"
  condition_query = ".query_result == {} or .query_result == [] or .query_result == null"

  value = "There are no elements"
}
```

## Deliverables
- the unit tests covering the new behavior

**Comment by dobarx** [comment]
Shouldn't this be outside of the content block? Like using something like `dynamic` block with a condition when to render content.

**Comment by traut** [comment]
with https://github.com/blackstork-io/fabric/issues/142 in mind, I think this issue still stands -- dynamic blocks are not well suitable for not-null checks

**Comment by dobarx** [comment]
@traut What if we added a child block type for all `content` blocks? Something like:
```hcl
content table {
  query = "..."
  ...
  empty_result {
    content text {
       value = "Empty data!"
    }
}
```
This would be a bit more flexible because we could specify any fallback content block.

**Comment by traut** [comment]
that's neat, it's declarative and provides more options. To reduce the nest-iness, maybe it can be a content block with some flag, like 

```hcl
content table {
  query = "..."
  ...
  when_empty content text {
     value = "Empty data!"
  }
}
```
?

**Comment by traut** [comment]
still not great :) it's confusing -- is `empty_result` / `when_empty` about the parent block or the nested block?

**Comment by dobarx** [comment]
`on_empty`? Could also extend in the future with fallbacks for errors `on_error`.

**Comment by traut** [comment]
oh, that's lovely.

```hcl
content table {
  query = "..."
  ...
  on_empty content text {
     value = "Empty data!"
  }
}

content table {
  query = "..."
  ...
  on_empty content ref {
     base = content.text.empty_data
  }
}
```
would be ok, I think

**Comment by traut** [comment]
moving this to draft since https://github.com/blackstork-io/fabric/issues/142 already can provide the functionality, even though a bit wordy
